internal/core/data/init.go

```diff
@@ -15,44 +16,50 @@ package data
 
 import (
 	"fmt"
+	"os"
+	"os/signal"
 	"sync"
+	"syscall"
 	"time"
 
 	"github.com/edgexfoundry/edgex-go/internal"
 	"github.com/edgexfoundry/edgex-go/internal/core/data/interfaces"
 	"github.com/edgexfoundry/edgex-go/internal/core/data/messaging"
 	"github.com/edgexfoundry/edgex-go/internal/pkg/config"
-	"github.com/edgexfoundry/edgex-go/internal/pkg/consul"
 	"github.com/edgexfoundry/edgex-go/internal/pkg/db"
 	"github.com/edgexfoundry/edgex-go/internal/pkg/db/mongo"
 	"github.com/edgexfoundry/edgex-go/internal/pkg/startup"
 	"github.com/edgexfoundry/go-mod-core-contracts/clients"
 	"github.com/edgexfoundry/go-mod-core-contracts/clients/logging"
 	"github.com/edgexfoundry/go-mod-core-contracts/clients/metadata"
 	"github.com/edgexfoundry/go-mod-core-contracts/clients/types"
-	"github.com/pkg/errors"
+	"github.com/edgexfoundry/go-mod-registry"
+	"github.com/edgexfoundry/go-mod-registry/pkg/factory"
 )
 
 // Global variables
 var Configuration *ConfigurationStruct
 var dbClient interfaces.DBClient
 var LoggingClient logger.LoggingClient
-var chEvents chan interface{} //A channel for "domain events" sourced from event operations
-var chConfig chan interface{} //A channel for use by ConsulDecoder in detecting configuration mods.
+var registryClient registry.Client
+var chEvents chan interface{}      //A channel for "domain events" sourced from event operations
+var errChannel chan error          //A channel for "config wait error" sourced from Registry
```

Comment:
We have two different formats for channel vars now -- `chEvents` which denotes a channel for Events and `errChannel` which denotes a channel for errors. My preference would be standardize these vars starting with `ch` followed by their intent -- `chEvents, chErr, chUpdates`

Reply:
I agree with standardizing, but my preference is intent followed by ```channel``` . :-) IMHO i make the code more readable.

Reply:
Take this into consideration here
https://dave.cheney.net/practical-go/presentations/qcon-china.html#_dont_name_your_variables_for_their_types

Reply:
Ha, I totally agree with this, mostly. I don't like the `config` example using `conf` or `c`, prefer `configuration`. IMHO using `ch` above is just a short hand for the channel type. So using this standard I would suggest the following names: `domainEvents`, `registryErrors` and `registryUpdates`. This more closely follows the `map` example. Thoughts?

Reply:
I think those naming suggestions are good. However I would do them as part of a follow-on refactoring PR so as not to lose focus of this PR.